src/meta/src/hummock/manager/commit_epoch.rs

```diff
@@ -163,204 +169,44 @@ impl HummockManager {
             .state_table_info
             .build_table_compaction_group_id();
 
-        let mut new_table_ids = None;
-
+        let mut new_table_ids = HashMap::new();
         // Add new table
         if let Some(new_fragment_table_info) = new_table_fragment_info {
-            let new_table_ids = new_table_ids.insert(HashMap::new());
             if !new_fragment_table_info.internal_table_ids.is_empty() {
-                for table_id in &new_fragment_table_info.internal_table_ids {
-                    if let Some(info) = new_version_delta
-                        .latest_version()
-                        .state_table_info
-                        .info()
-                        .get(table_id)
-                    {
-                        return Err(Error::CompactionGroup(format!(
-                            "table {} already exist {:?}",
-                            table_id.table_id, info,
-                        )));
-                    }
-                }
+                let new_table_id_to_cg = self.on_handle_add_new_table(
```

Comment:
We can pass `&mut new_table_ids` to `on_handle_add_new_table` and update it directly there.

The body of `on_handle_add_new_table` does not use `self`. We'd better declare the `on_handle_add_new_table` inside this `commit_epoch`, because it is only used here.